lib/rspec/active_model/mocks/mocks.rb

```diff
@@ -135,11 +135,16 @@ def self.param_delimiter; "-"; end
                 :blank? => false }.merge(stubs)
 
       double("#{model_class.name}_#{stubs[:id]}", stubs).tap do |m|
+        mock_store_klass = RSpec::ActiveModel::Mocks::Mocks
         if model_class.method(:===).owner == Module && !stubs.key?(:===)
+          allow(mock_store_klass).to receive(:mock_store) do
+            @mock_store ||= Hash.new { |h, k| h[k] = [] }
+          end
```

Comment:
```suggestion
          allow(model_class).to receive(:__rspec_mocks) do
            @mock_store ||= []
          end
```

Reply:
I think the `@mock_store` gets defined in the wrong context (RSpec ExampleGroup or Example), so it would need to change to e.g.
```rb
model_class.instance_variable_get(:@mock_store) ||
  model_class.instance_variable_set(:@mock_store, [])
```
But when I do that, I get specs failing with

> `#<Double "MockableModel_1037"> was originally created in one example but has leaked into another example and can no longer be used. rspec-mocks' doubles are designed to only last for one example, and you need to create a new one in each example you wish to use it for.`

Reply:
That raises a good point, as the context this is executing is the example, we can actually just define a variable without a mock and it will be cleaned up, I've pushed an alternative implementation to this PR

Reply:
That is neat, thanks @JonRowe. Feel free to squash the commits as you see fit.